google/cloud/storage/internal/grpc_hmac_key_metadata_parser.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_GRPC_HMAC_KEY_METADATA_PARSER_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_GRPC_HMAC_KEY_METADATA_PARSER_H
+
+#include "google/cloud/storage/hmac_key_metadata.h"
+#include "google/cloud/storage/version.h"
+#include "google/cloud/options.h"
+#include <google/storage/v2/storage.pb.h>
+
+namespace google {
+namespace cloud {
+namespace storage {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace internal {
+
+struct GrpcHmacKeyMetadataParser {
```

Comment:
Q: Do we like using `struct`s to only group free functions?

Reply:
It allows me to forward declare a `friend struct Foo;` this is used all over the place in `storage`. Maybe we should get rid of the need for friendship on the `*Metadata` classes, but that would be a larger refactoring, and there are too many functions to forward declare otherwise.